app/src/main/java/org/dhis2/data/forms/dataentry/fields/FormViewHolder.java

```diff
@@ -50,7 +51,7 @@ public FormViewHolder(ViewDataBinding binding) {
     public void initFieldFocus() {
         if (currentUid != null) {
             currentUid.observeForever(fieldUid -> {
-                if (Objects.equals(fieldUid, this.fieldUid)) {
+                if((selectedFieldUid == null && Objects.equals(fieldUid, this.fieldUid)) || ((selectedFieldUid != null) && selectedFieldUid.equals(this.fieldUid))){
```

Comment:
Too difficult to follow, could you extract it to a method call with a friendly name?